### DISCUSSION TITLE
I think Vite with Remix  didn't like tailwind integraiton?

### DISCUSSION THREAD

**Discussion by onesamket** [remix-run/remix#7851] [Q&A] [answered]
```
import stylesheet from "~/tailwind.css";
export const links = () => [  { rel: "stylesheet", href: stylesheet }];

```
Default and named imports from CSS files are deprecated. Use the ?inline query instead. For example: import stylesheet from "~/tailwind.css?inline"
  Plugin: vite:import-analysis

what is the issue?

**Comment by jrestall** [comment, marked as answer]
You'll want to switch to using side effect imports. https://remix.run/docs/en/main/future/vite#fix-up-css-imports

**Reply by manzaloros** [reply]
If you configure Tailwind this was, can you import the tailwind.css file in your root.tsx as a side effect, or do you now have to import the tailwind css per route (which wouldn’t make sense)?

**Reply by jrestall** [reply]
Just in the root.tsx is fine.

**Reply by manzaloros** [reply]
Thanks, @jrestall!

Tangentially related: do you happen to know how [CSS-in-JS](https://ant.design/docs/blog/hydrate-cssinjs) tools should be handled with Remix and Vite, e.g. [Ant Design v5](https://ant.design/docs/react/server-side-rendering)?

With previous Remix versions I was generating all the AntD CSS in a resource-route loader that I was using in the links in the root.tsx, [like this solution](https://stackoverflow.com/a/76358324).

[Vite looks like it has a way to handle CSS in JS with SSR](https://vite-plugin-ssr.com/css-in-js), but I don’t know how you would use this with Remix that uses streaming now.

**Reply by onesamket** [reply]
I got it thanks to all of you 🙏

**Reply by manzaloros** [reply]
@jrestall , thanks for your help.

On the official express remix vite template Tailwind works as expected.

However, using a custom server (package.json invokes `remix dev --manual -c 'node server.js'` for `yarn dev`) Tailwind, Vite and Remix 2.2.0 gives a FoUC.

**Reply by jrestall** [reply]
You shouldn't be mixing the remix cli and vite, if you want to use vite, you should fully migrate to the official express vite template rather than use `remix dev`.

**Reply by manzaloros** [reply]
Really? There's a section of the Remix Vite docs that specifically mentions custom servers: https://remix.run/docs/en/main/future/vite#migrating-from-a-custom-server

**Edit**
I see that in the docs they now just invoke `node server.js`, rather than `remix dev`. So I changed that in my code and still get the FoUC with Tailwind.

**Reply by jrestall** [reply]
Yes, the vite migration docs mention to change your package.json scripts to `"dev": "node ./server.mjs",` rather than using the old remix cli. You will need to follow the official vite express template and not use the remix cli.

**Reply by manzaloros** [reply]
> You will need to follow the official vite express template and not use the remix cli.

Got it, I followed the Remix docs, for example using `unstable_createViteServer()`, (haven't checked out the express vite template) and Vite is working with my custom server, but I'm still getting a FoUC with Tailwind and Vite (even though I'm using the same configs mentioned in the Remix Vite docs for Tailwind).

**Reply by jrestall** [reply]
Could be a browser extension causing hydration issues. I've found react canary 18.3 helps if it is.

**Comment by dkkazmierczak** [comment]
Here https://remix.run/docs/en/main/future/vite#fix-up-css-imports-referenced-in-links they say that you will need to add **?url** to your css imports. They showed an example I used:
```
-import styles from "~/styles/dashboard.css";
+import styles from "~/styles/dashboard.css?url";

export const links = () => {
  return [
    { rel: "stylesheet", href: styles }
  ];
}
```

and added postcss.config.mjs
```
export default {
  plugins: {
    tailwindcss: {},
  },
};
```

This has worked for me :)

**Reply by isc30** [reply]
this should be in the official documentation

**Reply by Shompi** [reply]
Absolutely
